wgengine/filter/match.go

```diff
@@ -72,13 +74,16 @@ func (m Match) String() string {
 	} else {
 		ds = "[" + strings.Join(dsts, ",") + "]"
 	}
-	return fmt.Sprintf("%v=>%v", ss, ds)
+	return fmt.Sprintf("%v%v=>%v", m.IPProto, ss, ds)
```

Comment:
For single-item ss, this will concat the protocol to an IP address without brackets. I don't think we depend on the representation anywhere, maybe just remove the special cases for len(srcs/dsts)==1?

Reply:
but IPProto is always in brackets, so it's not touching.